pkg/tfvars/baremetal/baremetal.go

```diff
@@ -0,0 +1,107 @@
+// Package baremetal contains bare metal specific Terraform-variable logic.
+package baremetal
+
+import (
+	"encoding/json"
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/hardware"
+	libvirttfvars "github.com/openshift/installer/pkg/tfvars/libvirt"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/pkg/errors"
+)
+
+type config struct {
+	LibvirtURI         string `json:"libvirt_uri,omitempty"`
+	IronicURI          string `json:"ironic_uri,omitempty"`
+	Image              string `json:"os_image,omitempty"`
+	ExternalBridge     string `json:"external_bridge,omitempty"`
+	ProvisioningBridge string `json:"provisioning_bridge,omitempty"`
+
+	// Data required for control plane deployment - several maps per host, because of terraform's limitations
+	Hosts         []map[string]interface{} `json:"hosts"`
+	RootDevices   []map[string]interface{} `json:"root_devices"`
+	Properties    []map[string]interface{} `json:"properties"`
+	DriverInfos   []map[string]interface{} `json:"driver_infos"`
+	InstanceInfos []map[string]interface{} `json:"instance_infos"`
+}
+
+// TFVars generates bare metal specific Terraform variables.
+func TFVars(libvirtURI, ironicURI, osImage, externalBridge, provisioningBridge string, platformHosts []*baremetal.Host, image baremetal.Image) ([]byte, error) {
+	osImage, err := libvirttfvars.CachedImage(osImage)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to use cached libvirt image")
+	}
+
+	var hosts, rootDevices, properties, driverInfos, instanceInfos []map[string]interface{}
+
+	for _, host := range platformHosts {
+		// Get hardware profile
+		if host.HardwareProfile == "default" {
```

Comment:
Is it wrong that I don't like the defaulting here.  Why didn't we set it to hardware.DefaultProfileName back where we set it to "default" ?

Reply:
I guess this handles the case where someone explicitly put `default` in install-config

Reply:
`hardware.DefaultProfileName` is "unknown", which looks weird to me in the install-config.  Like @russellb mentioned on slack, we're still trying to understand the best way to represent hardware profiles so I'd expect this to change.